### ISSUE TITLE
can you all get paper's result?

### ISSUE THREAD

**Issue by zkm98** [ma-xu/pointMLP-pytorch#1]
I tried to collect deep learning on point cloud, I tried to run the model in my env, however I just got 92.99 acc on test set which is also a high score, but I cant get what paper says. So what's the reason of this phenomena

**Comment by zkm98** [comment]
It seems it's hard to train for me...

**Comment by 13952522076** [comment]
@zkm98 

I suppose you are talking about the performance on ModelNet40. The performance on ModelNet40 of almost all methods are not stable, see (https://github.com/CVMI-Lab/PAConv/issues/9#issuecomment-873371422).
If you run the same codes for several times, you will get different results.

Here is the log we got the testing **94.5** acc with voting: https://drive.google.com/file/d/1z-QbWLMlGxfMVPImcLeaDOVOJGvm64ny/view?usp=sharing

Here is the log we got the training **94.1** acc: https://drive.google.com/file/d/1T8ZljhO2evsxQ0aL7l0xbd_dVHgargzT/view?usp=sharing

Also, the randomness of ModelNet40 is our motivation to experiment on ScanObjectNN benchmark, and report the mean/std results of several runs.

Please let me know if you have any questions or concerns.

Best,
Xu

**Comment by 13952522076** [comment]
> It seems it's hard to train for me...

@zkm98 Could you please clarify what do you mean by "hard to train"?
Empirically, pointmlp is easy to train in terms of deployment, speed,  and loss curve, etc.

**Comment by zkm98** [comment]
about hard to train, I mean, it is hard to get the results in your papers, sorry that I dont clearly define it. For most experiments I trained, I just got the result of 92.8% using pointmlp Elite, I was trying to train from scratch, although maybe I can get the result as you say using the pretrained models. It cant talk me into it.

**Comment by ma-xu** [comment]
@zkm98 I uploaded checkpoint/logs for PointMLP and PointMLP-elite (also, the bug-fixed results and checkpoints are also uploaded). Please see the README file.

As I mentioned previously, the results on ModelNet40 are not stable (but always greater than 93.x for both pointmlp and pointmlp-elite). The randomness is the motivation for us to consider ScanObjectNN benchmark.

**Comment by le-cheng** [comment]
Thank you very much for your wonderful work. I tried testing with your pre-trained model and with the best parameters I only got 93.6. I would like to know what this deals with, theoretically with the same model, using the same parameters, with only validation I should get 94.1.![image](https://user-images.githubusercontent.com/48431020/157153731-303a69fc-a18f-45f7-bf80-4091ad733e36.png)

**Comment by ma-xu** [comment]
@le-cheng Thanks for your interest.  I will double-check and reply to you by  Wednesday.

**Comment by ma-xu** [comment]
@le-cheng 
I just double-checked the testing result, there is no problem, and I can get the same results as I claimed. See screenshot.
![3681646765605_ pic](https://user-images.githubusercontent.com/6723062/157306065-1aacec68-60b5-43f9-a927-e10d950aa83a.jpg)
For your convenience,  I uploaded a simple test code.

**Comment by le-cheng** [comment]
@ma-xu 
OK, my problem has been solved！
![image](https://user-images.githubusercontent.com/48431020/157375863-f27eb065-aa1e-4971-aa25-57e70bca749a.png)
I would like to ask if your training method is just multiple training sessions, or is there some other parameter adjustment. Am I using your code under what circumstances it is possible to get the same results as you.

**Comment by ma-xu** [comment]
@le-cheng  Sry that I did not understand you question. If you can talking about the unstable peroformance?
I trained with the default parameters (see log files).

The performance on ModelNet40 of almost all methods are not stable, see (https://github.com/CVMI-Lab/PAConv/issues/9#issuecomment-873371422).
If you run the same codes for several times, you will get different results.

Also, the randomness of ModelNet40 is our motivation to experiment on ScanObjectNN benchmark, and to report the mean/std results of several runs.

**Comment by le-cheng** [comment]
Well, thanks.

**Comment by coallar** [comment]
![image](https://user-images.githubusercontent.com/66529823/159208112-9ddb3cb3-cec9-441f-94d5-fe508195b034.png)

only 93.071 can I get！

**Comment by ma-xu** [comment]
@coallar 

COULD YOU PLEASE READ WHAT I POSTED PREVIOUSLY IN THIS ISSUE?

Again, the performance on ModelNet40 is not deterministic，the perofmrance has a large variance. This also motivates us to  consider ScanObjectNN as a benchmark. Please see above and the paper.

**Comment by yanghu819** [comment]
For the experiment itself is unstable, could you produce multi-times running logs or the exact random seed to reproduce the best checkpoint.

**Comment by ma-xu** [comment]
@yanghu819 
Sure, I upload the 6 runs of PointMLP (last_checkpoints is removed to save space, model names is our fixed developing name, ignore it). Here is the link: https://drive.google.com/drive/folders/1SUuMDjaWhxURaUO7VtPpJ_vnxvaBA9eO?usp=sharing
The folder name pattern is [modelname]-[time]-[randgeneratedseed]. That is, we train the model (bug fixed version, we can also provide unfixed version for the submission if needed) on Feb/09/2022, with 6 randomly generated seeds. 

Train with the exactly same seed, one propobaly still cannot get the deterministic result, we tested that. You can also have a try, the seed is provided in the folder name.

Besides, a good choice is to report the mean and std of several runs as in Table 3 on ScanObjectNN benchmark. You can easily get a better PointMLP result than we reported.

**Comment by ma-xu** [comment]
@zkm98 @yanghu819 @le-cheng @coallar 
I will close this issue since there is no further discussion. Feel free to reopen it if necessary.
